### PR TITLE
Cardboard cutouts are now tactical.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Basically, this means players holding cardboard cutouts will now assume their appearance, just like for potted plants. Good for pranking.

I've had to tweak the tactical component and the waddling element a bit to get them to work as intended while dealing with the multiple sources of the waddling element. 

## Changelog

:cl:
add: Players holding cardboard cutouts will now assume their appearance, just like for potted plants.
/:cl:

### CODE REVIEW

**Comment by Likteer** [comment]
Could you please make the sprite waddle, like in the Tom Cruise vid?

**Comment by Ghommie** [comment]
I checked having both the waddling with the "tactical" appearance at once, but the two don't mix well together. In fact, the overriding, tactical imagery of the item doesn't respect the transformation matrix of the mob, meaning it won't skew, rotate, stretch or scale, but it'll still bounce up and down like it's hopping, rather than waddling.

That said, I'll get it done.

**Comment by Loafin34** [comment]
Imagine clownops using the cardboard cutouts to disguise themselves as cultists.

**Comment by Ghommie** [comment]
Ok, I've had to do a few tweaks to implement waddling and make sure the cardboard appearances always faced south when moving, also scratching most of the dumb changes I've done to the tactical element. It works great now.

**Review by Likteer — APPROVED** [review]
Magnificent PR, a victory for good taste

File: code/datums/elements/_element.dm [hunk 1]
```diff
@@ -75,3 +75,28 @@
...
+ */
+/datum/proc/AddElementTrait(trait, source, datum/element/eletype, ...)
+	if(!ispath(eletype, /datum/element))
+		CRASH("AddElementTrait called, but [eletype] is not of a /datum/element path")
+	var/already_had_trait = HAS_TRAIT(src, trait)
+	ADD_TRAIT(src, trait, source)
+	if(already_had_trait)
+		return
+	var/list/arguments = list(eletype)
+	if(length(args) > 3)
```
Comment (SuperNovaa41) [inline code/datums/elements/_element.dm:93]: magic number, what does this mean

File: code/game/objects/items/cardboard_cutouts.dm [hunk 2]
```diff
@@ -42,12 +46,22 @@
...
+	. = ..()
+	REMOVE_TRAIT(user, TRAIT_WADDLING, REF(src))
+
 /obj/item/cardboard_cutout/proc/push_over()
 	appearance = initial(appearance)
 	desc = "[initial(desc)] It's been pushed over."
 	icon_state = "cutout_pushed_over"
 	remove_atom_colour(FIXED_COLOUR_PRIORITY)
 	pushed_over = TRUE
+	qdel(GetComponent(/datum/component/tactical))
```
Comment (SuperNovaa41) [inline code/game/objects/items/cardboard_cutouts.dm:64]: hey man whats the big idea

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: code/datums/elements/_element.dm (hunk 1 above)
Comment (Ghommie) [reply code/datums/elements/_element.dm:93]: It really isn't all that magical, just look at the number of fixed args. I'll give it a comment to explain though.

File: code/game/objects/items/cardboard_cutouts.dm [hunk 3]
```diff
@@ -179,19 +197,24 @@
...
 /// This proc sets the cardboard cutout item's vars.
 /datum/cardboard_cutout/proc/apply(obj/item/cardboard_cutout/cutouts)
+	if(isnull(applied_appearance))
+		applied_appearance = image(fcopy_rsc(getFlatIcon(preview_appearance, no_anim = TRUE)))
+	applied_appearance.plane = cutouts.plane
+	applied_appearance.layer = cutouts.layer
 	cutouts.appearance = applied_appearance
 	cutouts.name = get_name()
 	cutouts.desc = applied_desc
+	SEND_SIGNAL(cutouts, COMSIG_CARDBOARD_CUTOUT_APPLY_APPEARANCE, src)
```
Comment (Jacquerel) [inline code/game/objects/items/cardboard_cutouts.dm:217]: Don't like this signal, smells like a proc call.
Good news though, it also seems kind of pointless? Why add another signal here when the only place you want to trigger would also be triggered by calling `cutouts.update_icon(NONE)`?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**